### PR TITLE
fix: Remove explicit dependency

### PR DESCRIPTION
**Pull request by jesse-c**
It's not directly used. Relax the constraint, so the package can be used elsewhere.

Update: Let the resolver work it out.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def readme():
               'dill>=0.3.0, <0.4.0',
               'transformers>=4.7.0, <5.0.0',
               'tqdm>=4.28.1, <5.0.0',
-              'pydantic <2.0.0',
+              'pydantic>=1.8.0, <3.0.0',
```
[the remark this answers is not in the document]
Comment (jesse-c) [reply setup.py:61]: Ah, that makes a bit more sense, thank you! I don't think there's a reason for the lower-bound, so I'll remove it and see how it fairs.

[the remark this answers is not in the document]
Comment (jesse-c) [reply setup.py]: ✅

File: setup.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def readme():
               'dill>=0.3.0, <0.4.0',
               'transformers>=4.7.0, <5.0.0',
               'tqdm>=4.28.1, <5.0.0',
-              'pydantic <2.0.0',
+              'pydantic<3.0.0',
```
Comment (sakoush) [inline setup.py:61]: I dont think we even need to specify pydantic here as it is a sub dependency of ray and not related to alibi

Comment (jesse-c) [reply setup.py]: ✅

**Pull request merged**